src/NHibernate.Validator/Event/ValidatePreInsertEventListener.cs

```diff
@@ -46,7 +46,7 @@ public void Initialize(Configuration cfg)
 				Engine = Environment.SharedEngineProvider.GetEngine();
 			}
 
-			IEnumerable<PersistentClass> classes = cfg.ClassMappings.Where(clazz => clazz.ClassName != null).Where(clazz => !clazz.ClassName.StartsWith("NHibernate.Envers"));
+			IEnumerable<PersistentClass> classes = cfg.ClassMappings.Where(clazz => clazz.ClassName != null && clazz.MappedClass != null);

```

Comment:
I would tend to change that to `cfg.ClassMappings.Where(clazz => clazz.HasPocoRepresentation);`

Testing both `ClassName` and `MappedClass` is redundant, `MappedClass` reflect `ClassName`. (And `HasPocoRepresentation` is just `ClassName != null`, but with a more meaningful naming.)

The whole old test was added in 6449be36d129881045e0e0527ce3ac97b38ff1b5, I would like to know why filtering manually envers classes was considered required. If not filtering it may cause failure, we need a test for that. @TimurZanagar, can you tell us?